nixos/modules/services/network-filesystems/iroh.nix

```diff
@@ -0,0 +1,275 @@
+  options = {
+    services.iroh = {
+
+      # By convention, modules have an "enable" option that determines whether they are active or
+      # not.
+      enable = mkEnableOption "Iroh is a protocol for syncing bytes. Bytes of any size, across any number of devices. Use iroh to create apps that scale different.";
+
+      package = mkPackageOption pkgs "iroh" { };
+
+      # NB: This is READ ONLY because there's no way to configure the listen port; iroh exposes as
+      # few configuration knobs as possible. The default port can be read from this read-only
+      # option, but if a different port is chosen by the service it has to be retrieved another way (?).
+      listenPort = mkOption {
+        description = "Address iroh listens on for connections from other nodes. If the port is taken iroh will choose a random port to listen on, so it can't be configured. This port is NOT automatically opened in the firewall.";
+        type = types.port;
+        readOnly = true;
+        default = listenPort;
+      };
+
+      rpcPort = mkOption {
+        description = "Port for localhost-only Remote Procedure Calls, used to control an iroh node from another process. If the port is taken iroh will fail to start. This port is NOT automatically opened in the firewall.";
+        type = types.port;
+        readOnly = true;
+        default = rpcPort;
+      };
+
+      metricsPort = mkOption {
+        description = "Port at which metrics data is served. By default it is not enabled. Value must be -1 to disable serving metrics, or a valid port number. This port is NOT automatically opened in the firewall.";
+        type = types.either (types.ints.between (-1) (-1)) types.port;
+        default = metricsPort;
+      };
+
+      noTicket = mkOption {
+        description = "Do not print the all-in-one ticket to get the data from this node.";
+        type = types.bool;
+        default = false;
+      };
+
+      tag = mkOption {
+        description = "An optional tag to apply to hosted data.";
+        type = types.nullOr types.str;
+        default = null;
+      };
+
+      extraFlags = mkOption {
+        type = types.listOf types.str;
+        description = "Extra flags passed to the Iroh daemon";
+        default = [ ];
+      };
+
+      fileLogs = mkOption {
+        description = "Set the output location for logs.";
+        type = logOutput;
+      };
+
+      relayNodes = mkOption {
+        description = "Nodes used to assist in holepunching when NAT traversal fails. If configured, these nodes replace the default set of relay nodes.";
+        type = types.listOf relayNode;
+        default = [
+          {
+            url = "https//use1-1.derp.iroh.network.";
+            stunOnly = false;
+            stunPort = 3478;
+          }
+          {
+            url = "https//euw1-1.derp.iroh.network.";
+            stunOnly = false;
+            stunPort = 3478;
+          }
+        ];
+      };
+
+      extraRelayNodes = mkOption {
```

Comment:
currently this is simply added to the other nodes - will it serve an additional purpose in the future?